core/src/main/java/org/mskcc/cbio/portal/authentication/aaf/PortalUserDetailsService.java

```diff
@@ -0,0 +1,149 @@
+
+import java.util.List;
+
+/**
+ * Responsible for verifying that an Australian Access Federation (AAF) 
+ * name has been registered in the portal database.
+ * 
+ * Implementation based on code in googleplus PortalUserDetailsService
+ * 
+ * @author shaun-intersect
+ */
+public class PortalUserDetailsService implements UserDetailsService {
+
+    private static final Log log = LogFactory.getLog(PortalUserDetailsService.class);
+
+    // ref to our user dao
+    private final PortalUserDAO portalUserDAO;
+
+    private final String sharedSecret;
+    private final String primaryURL;
+    private final Boolean productionFederation;
+    
+    /**
+     * Constructor.
+     * <p>
+     * Takes a ref to PortalUserDAO used to authenticate registered users in the
+     * database.
+     *
+     * Also takes details used as part of service registration of the application with AAF Rapid Connect, specifically 
+     * the secret that was shared with Rapid Connect to sign the JWT and the web accessible endpoint that Rapid Connect 
+     * sends assertion header to as part of the HHTPS POST request.
+     *
+     * @param portalUserDAO ref to PortalUserDAO used to authenticate registered users in the database
+     * @param sharedSecret secret shared between cBioPortal and AAF Rapid Connect for signing and verifying JWT
+     * @param primaryURL HTTPS endpoint that accepts the POST request containing the assertion parameter
+     * @param productionFederation true if using the AAF productionFederation, false if using the test federation
+     */
+    public PortalUserDetailsService(PortalUserDAO portalUserDAO, String sharedSecret, String primaryURL, 
+                                    Boolean productionFederation) {
+        this.portalUserDAO = portalUserDAO;
+        this.sharedSecret = sharedSecret;
+        this.primaryURL = primaryURL;
+        this.productionFederation = productionFederation;
+    }
+    
+    @Override
+    public UserDetails loadUserByUsername(String assertion) throws UsernameNotFoundException {
+        String username = null;
+        try {
+            AafJwtHandler jwtHandler = new AafJwtHandler(assertion, sharedSecret, primaryURL, productionFederation);
+            username = jwtHandler.getEmail();
+        } catch (Exception e) {
+            log.debug("Exception occurred whilst handling assertion: " + e + '\n' + ExceptionUtils.getStackTrace(e));
+        }
+                
+        // set the username into the global state so other components can find out who
+        // logged in or tried to log in most recently
+        DynamicState.INSTANCE.setCurrentUser(username);
+        if (log.isDebugEnabled()) {
+            log.debug("loadUserByUsername(), attempting to fetch portal user, email: " + username);
+        }
+        PortalUserDetails toReturn = null;
+        User user = null;
+        try {
+            user = portalUserDAO.getPortalUser(username);
```

Comment:
Apart from lines 96-101 this class is a duplicate of the [google authentication one](https://github.com/IntersectAustralia/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/portal/authentication/googleplus/PortalUserDetailsService.java). It doesn't look like the [implementing class](https://github.com/IntersectAustralia/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/portal/dao/internal/PortalUserJDBCDAO.java) does, I would have assumed an SQL exception may have been potentially thrown at some point but it doesn't seem thats the case. I'm not sure why this would have been included in the original version.